### ISSUE TITLE
Seamless Integration between Collector and Trek2There

### ISSUE THREAD

**Issue by craigcarps** [Esri/Trek2There#47]
User feedback has strongly indicated the need to launch Trek2There directly from Collector when the user selects an existing Collector point or location. This would then pass the coordinate values from Collector to Trek2There and allow for navigation to the point. When the user arrives at the point the Collector app re-opens.

**Comment by jakc** [comment]
Agree for the need in Collector Craig.  You are probably already aware, [but there is a sample template](https://geonet.esri.com/groups/survey123/blog/2016/10/24/introducing-trek2there) for Survey123 that works nicely between the two apps.

**Comment by craigcarps** [comment]
Hi Simon

Yeah I've seen the template with survey 123, the problem I have is a few
clients who are already using collector and want the off road navigation
capability.

Thanks for the link

Craig

**Comment by Lmawhinney7175** [comment]
I agree with the need in Collector Craig as well.  I would really like to be able to select the point I want to go to and then get feed back to how close I am.  This would be valuable.

**Comment by IsmaelInRedlands** [comment]
Hi! We are going to add integration of Trek2There with Collector in a couple of phases:
1) December 13: The 4.4 update of ArcGIS Online will let you create Web Map popups including links to Trek2There. We will follow a pattern similar to what is described here for  integrating Collector with Survey123.
2) March update of ArcGIS Online: As of December  13, you will always need to have in your Collector features the X,Y values as attributes. Otherwise, you will not be able to pass the destination to Trek2There dynamically.  In March, tentatively, we will enhance Web Map popups so the XY coordinates can be pulled dynamically from the geometry of the feature: you will not need to have two pre-calculated XY fields with the coordinates.  This is good, because it means the XY attributes will always be in sync with the coordinates stored in the SHAPE field.

**Comment by Lmawhinney7175** [comment]
Hello,


Thank you for forwarding this information.  I look forward to the updates.


Laura

>>> Ismael Chivite <notifications@github.com> 11/21/2016 1:31 PM >>>

 
Hi! We are going to add integration of Trek2There with Collector in a couple of phases:
December 13: The 4.4 update of ArcGIS Online will let you create Web Map popups including links to Trek2There. We will follow a pattern similar to what is described here for integrating Collector with Survey123.
March update of ArcGIS Online: As of December 13, you will always need to have in your Collector features the X,Y values as attributes. Otherwise, you will not be able to pass the destination to Trek2There dynamically. In March, tentatively, we will enhance Web Map popups so the XY coordinates can be pulled dynamically from the geometry of the feature: you will not need to have two pre-calculated XY fields with the coordinates. This is good, because it means the XY attributes will always be in sync with the coordinates stored in the SHAPE field.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or mute the thread.


  
 {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/Esri/Trek2There","title":"Esri/Trek2There","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/Esri/Trek2There"}},"updates":{"snippets":[{"icon":"PERSON","message":"@IsmaelInRedlands in #47: Hi! We are going to add integration of Trek2There with Collector in a couple of phases:\r\n1) December 13: The 4.4 update of ArcGIS Online will let you create Web Map popups including links to Trek2There. We will follow a pattern similar to what is described here for integrating Collector with Survey123.\r\n2) March update of ArcGIS Online: As of December 13, you will always need to have in your Collector features the X,Y values as attributes. Otherwise, you will not be able to pass the destination to Trek2There dynamically. In March, tentatively, we will enhance Web Map popups so the XY coordinates can be pulled dynamically from the geometry of the feature: you will not need to have two pre-calculated XY fields with the coordinates. This is good, because it means the XY attributes will always be in sync with the coordinates stored in the SHAPE field."}],"action":{"name":"View Issue","url":"https://github.com/Esri/Trek2There/issues/47#issuecomment-262058230"}}}

**Comment by jdzurisin** [comment]
Thanks so much for the application itself and for the gameplan to integrate to Collector. For our team, and many others like it, this has been a crucial limitation of Collector to date. With seamless integration of Trek2There in a similar vein to what has been done with Navigator this would finally be a perfect ecosystem for our crews. Really looking forward to seeing this come to fruition.

**Comment by craigcarps** [comment]
Ismael

That's brilliant news, an early Xmas present for clients :)

The use of the xy coordinates direct from the geometry will be exactly what is needed.

Cheers for the update.
Craig

**Comment by Lmawhinney7175** [comment]
Hello Ismael,
 
Can you please elaborate on how the location data is passed from the
web map to the trek2there app?
 
Kind Regards,
 
Laura

Laura Mawhinney
GIS Technician
CITY OF TUCSON
Tucson Water
(520) 837-2175
 
"Data is important... However, not as important as the answers it
provides or the stories that it tells."
 

 >>> Ismael Chivite <notifications@github.com> 11/21/2016 1:31 PM >>>
Hi! We are going to add integration of Trek2There with Collector in a
couple of phases:
	1.  December 13: The 4.4 update of ArcGIS Online will let you
create Web Map popups including links to Trek2There. We will follow a
pattern similar to what is described here for integrating Collector with
Survey123.
	2.  March update of ArcGIS Online: As of December 13, you will
always need to have in your Collector features the X,Y values as
attributes. Otherwise, you will not be able to pass the destination to
Trek2There dynamically. In March, tentatively, we will enhance Web Map
popups so the XY coordinates can be pulled dynamically from the geometry
of the feature: you will not need to have two pre-calculated XY fields
with the coordinates. This is good, because it means the XY attributes
will always be in sync with the coordinates stored in the SHAPE field.
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
( https://github.com/Esri/Trek2There/issues/47#issuecomment-262058230)
, or mute the thread
(

**Comment by IsmaelInRedlands** [comment]
@Lmawhinney7175 @craigcarps @jdzurisin @jakc  Now that ArcGIS Online 4.4 is out, I added some details regarding integration of Trek2There with Collector and Explorer. Find out more at the ArcGIS blog: https://blogs.esri.com/esri/arcgis/2016/12/22/introducing-trek2there/

**Comment by marikavertzonis** [comment]
- Tested across devices with the new ArcGIS Online features - there has been some improvement on some devices. In short - you can launch Trek2There on on Windows and Android - you cant automatically go back.
 - Tested on windows 10, ios 8.4 (iphone 6), ios 9.3 (iphone 5), android 6.0.1 (samsung s6), android 7 (nexus 6)
    - On Windows, successful can launch Trek2There from Collector using a popup configured as per the blog article above. When you reach destination you must close Trek2There and switch back to Collector - no automatic return (known issue - requires change in AppStudio)
    - On Android, successful can launch Trek2There from Collector using a popup configured as per the blog article above. Coordinates appear to be switched (the destination that should be 80m is 14,000km). When you click End, you don't get taken back to Collector, stays on 'No destination set' page of Trek2There.
    - On iOS - Trek2There doesn't launch from the Collector pop-up. Instead you get an empty looking window that has 'Done' and 'Open in Safari' in the title area.

**Comment by MartinLauenborg** [comment]
@IsmaelInRedlands 
Do you or anyone else know if the popups has been enahnced with the oppotunity to dynamically use geometry in the March 2017 release?

>  2.  March update of ArcGIS Online: As of December 13, you will
always need to have in your Collector features the X,Y values as
attributes. Otherwise, you will not be able to pass the destination to
Trek2There dynamically. In March, tentatively, we will enhance Web Map
popups so the XY coordinates can be pulled dynamically from the geometry
of the feature: you will not need to have two pre-calculated XY fields
with the coordinates. This is good, because it means the XY attributes
will always be in sync with the coordinates stored in the SHAPE field.

**Comment by TViveiros** [comment]
Hello...any idea when the issue launching Trek2There from Collector on iOS will be resolved?
Thanks.
Tony
